### PR TITLE
Replace lwz with rlwinm instead of mr

### PR DESCRIPTION
**Pull request by AgostonSzepessy**
In some cases the compiler was optimizing away lwz, which was being used
to clear the high 32 bits of a variable, resulting in garbage values.
Replacing lwz with rlwinm solves this.

### CODE REVIEW

File: compiler/p/codegen/OMRCodeGenerator.cpp [hunk 1]
```diff
@@ -1229,11 +1232,23 @@ static void lhsPeephole(TR::CodeGenerator *cg, TR::Instruction *storeInstruction
    // Found the pattern:
    //   stw/std rX, ...
    //   lwz/ld rY, ...
-   // and will remove lwz/ld, which will result in:
-   //   stw/std rX, ...
+   // and will remove lwz, which will result in:
+   //   stw rX, ...
+   //   rlwinm rY, rX, 0, 0xffffffff
+   // and then the rlwinm peephole should run on the resulting rlwinm
```
Comment (ymanton) [inline compiler/p/codegen/OMRCodeGenerator.cpp]: Get rid of this comment line, unlike the case for `mr`, there is no `rlwinm` peephole.

File: compiler/p/codegen/OMRCodeGenerator.cpp [hunk 2]
```diff
@@ -1214,10 +1214,13 @@ static void lhsPeephole(TR::CodeGenerator *cg, TR::Instruction *storeInstruction
       //   lwz/ld rX, ...
       // will remove ld
       // and replace lwz with rlwinm, rX, rX, 0, 0xffffffff
-      if (loadInstruction->getOpCodeValue() == TR::InstOpCode::lwz && performTransformation(comp, "O^O PPC PEEPHOLE: Replace lwz " POINTER_PRINTF_FORMAT " with rlwinm after store " POINTER_PRINTF_FORMAT ".\n", loadInstruction, storeInstruction))
+      if (loadInstruction->getOpCodeValue() == TR::InstOpCode::lwz)
```
Comment (ymanton) [inline compiler/p/codegen/OMRCodeGenerator.cpp]: Can you get rid of the trailing space here?

**Comment by AgostonSzepessy** [comment]
@ymanton I've made the changes. Could you please look it over again?

**Comment by ymanton** [comment]
LGTM, thanks.

**Pull request merged**